src/order/RelayOrder.ts

```diff
@@ -0,0 +1,308 @@
+import { SignatureLike } from "@ethersproject/bytes";
+import {
+  PermitBatchTransferFrom,
+  PermitBatchTransferFromData,
+  SignatureTransfer,
+  Witness,
+} from "@uniswap/permit2-sdk";
+import { BigNumber, ethers } from "ethers";
+
+import { PERMIT2_MAPPING } from "../constants";
+import { MissingConfiguration } from "../errors";
+import { ResolvedRelayOrder } from "../utils/OrderQuoter";
+import { getDecayedAmount } from "../utils/dutchDecay";
+
+import { Order, OrderInfo, OrderResolutionOptions } from "./types";
+
+export type RelayInput = {
+  readonly token: string;
+  readonly recipient: string;
+  readonly startAmount: BigNumber;
+  readonly maxAmount: BigNumber;
+};
+
+export type RelayInputJSON = Omit<RelayInput, "startAmount" | "maxAmount"> & {
+  startAmount: string;
+  maxAmount: string;
+};
+
+type RelayOrderNestedOrderInfo = Omit<
+  OrderInfo,
+  "additionalValidationContract" | "additionalValidationData"
+>;
+
+export type RelayOrderInfo = RelayOrderNestedOrderInfo & {
+  inputs: RelayInput[];
+  decayStartTime: number;
+  decayEndTime: number;
+  actions: string[];
+};
+
+export type RelayOrderInfoJSON = Omit<RelayOrderInfo, "nonce" | "inputs"> & {
+  nonce: string;
+  actions: string[];
+  inputs: RelayInputJSON[];
+};
+
+type WitnessInfo = {
+  reactor: string;
+  swapper: string;
+  startAmounts: BigNumber[];
+  recipients: string[];
+  decayStartTime: number;
+  decayEndTime: number;
+  actions: string[];
+};
+
+const RELAY_ORDER_TYPES = {
```

Comment:
Maybe this should be RELAY_WITNESS_TYPES? Separate from the RelayOrder that is passed in to execute